hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/api/Query.kt

```diff
@@ -40,6 +39,32 @@ data class Query(
         }
       }
 
+  private fun BoolQueryBuilder.addAndQueries(attributes: Attributes) {
+    // matchers that aren't for nested fields can be added directly to the bool query
+    matchers
+      ?.findMatchers(attributes, nested = false)
+      ?.forEach { must(it.buildQuery(attributes)) }
+
+    // matchers for nested fields need to be grouped into nested queries
+    matchers
+      ?.findMatchers(attributes, nested = true)
+      ?.groupBy { it.attribute.substringBefore(".") }
+      ?.forEach { (prefix, matchers) ->
+        must(
+          QueryBuilders.nestedQuery(
+            prefix,
+            matchers.fold(QueryBuilders.boolQuery()) { boolQueryBuilder, typeMatcher ->
+              boolQueryBuilder.must(typeMatcher.buildQuery(attributes))
+            },
+            ScoreMode.None,
+          ),
+        )
+      }
```

Comment:
Any matchers for the same nested objects are grouped together into a NestedQuery which tells OpenSearch that all the matchers must match within the same object.